text/0617-cloudfront-oac-l2.md

```diff
@@ -0,0 +1,805 @@
+OAC is not available in China regions.
+
+#### New `OriginAccessControl` L2 Construct
+
+The OAC class for each origin type will extend a base class `OriginAccessControlBase` and set the value of `originAccessControlOriginType` accordingly.
+
+```ts
+/**
+ * Represents a CloudFront Origin Access Control
+ */
+export interface IOriginAccessControl extends IResource {
+  /**
+   * The unique identifier of the origin access control.
+   * @attribute
+   */
+  readonly originAccessControlId: string;
+}
+
+/**
+ * Common properties for creating a Origin Access Control resource.
+ */
+export interface OriginAccessControlBaseProps {
+  /**
+   * A description of the origin access control.
+   * @default - no description
+   */
+  readonly description?: string;
+  /**
+   * A name to identify the origin access control. You can specify up to 64 characters.
+   * @default - a generated name
+   */
+  readonly originAccessControlName?: string;
+  /**
+   * Specifies which requests CloudFront signs and the signing protocol.
+   * @default SIGV4_ALWAYS
+   */
+  readonly signing?: Signing;
+}
+
+/**
+ * Properties for creating a Origin Access Control resource.
+ */
+export interface S3OriginAccessControlProps extends OriginAccessControlBaseProps {}
+
+/**
+ * Origin types supported by Origin Access Control.
+ */
+export enum OriginAccessControlOriginType {
+  /**
+   * Uses an Amazon S3 bucket origin.
+   */
+  S3 = 's3',
+  /**
+   * Uses a Lambda function URL origin.
+   */
+  LAMBDA = 'lambda',
+  /**
+   * Uses an AWS Elemental MediaStore origin.
+   */
+  MEDIASTORE = 'mediastore',
+  /**
+   * Uses an AWS Elemental MediaPackage v2 origin.
+   */
+  MEDIAPACKAGEV2 = 'mediapackagev2',
+}
+
+/**
+ * Options for which requests CloudFront signs.
+ * Specify `always` for the most common use case.
+ */
+export enum SigningBehavior {
+  /**
+   * Sign all origin requests, overwriting the Authorization header
+   * from the viewer request if one exists.
+   */
+  ALWAYS = 'always',
+  /**
+   * Do not sign any origin requests.
+   * This value turns off origin access control for all origins in all
+   * distributions that use this origin access control.
+   */
+  NEVER = 'never',
+  /**
+   * Sign origin requests only if the viewer request
+   * doesn't contain the Authorization header.
+   */
+  NO_OVERRIDE = 'no-override',
+}
+
+/**
+ * The signing protocol of the Origin Access Control.
+ */
+export enum SigningProtocol {
+  /**
+   * The AWS Signature Version 4 signing protocol.
+   */
+  SIGV4 = 'sigv4',
+}
+
+/**
+ * Options for how CloudFront signs requests.
+ */
+export class Signing {
+  /**
+   * Sign all origin requests using the AWS Signature Version 4 signing protocol.
+   */
+  public static readonly SIGV4_ALWAYS = new Signing(SigningProtocol.SIGV4, SigningBehavior.ALWAYS);
+  /**
+   * Do not sign any origin requests.
+   */
+  public static readonly SIGV4_NEVER = new Signing(SigningProtocol.SIGV4, SigningBehavior.NEVER);
```

Comment:
If this is never, the signing method doesn't matter so this can just be `NEVER` or `NONE`

Reply:
Makes sense, updated to `NEVER` to be consistent with CloudFormation property